### PR TITLE
Implemented reset.

### PR DESCRIPTION
**Pull request by emilwestergren**
Recycled buffers are now cleared when resolution is changed.

@sebastianbaginski peer

### CODE REVIEW

File: HAL/HAL3/VideoProcessor.cpp [hunk 1]
```diff
@@ -40,12 +40,20 @@ VideoProcessor::~VideoProcessor() {
 	pthread_mutex_destroy(&this->metaMutex);
 	doubleBuffers.clear();
 }
+void VideoProcessor::reset() {
+	this->cameraBuffers.clear();
+	for(int i = 0; i < 4; i++) {
```
Comment (sebastianbaginski) [inline HAL/HAL3/VideoProcessor.cpp]: Obvious question, why 4 ?

Comment (emilwestergren) [inline HAL/HAL3/VideoProcessor.cpp]: Everyone knows 4 is the optimal size of a recycled buffer list...

Comment (sebastianbaginski) [inline HAL/HAL3/VideoProcessor.cpp]: Yes of course, I was just checking if you know that :+1:

**Comment by emilwestergren** [comment]
Updated.

**Comment by sebastianbaginski** [comment]
OK

**Pull request merged**